src/rdb.c

```diff
@@ -2178,15 +2193,26 @@ robj *rdbLoadObject(int rdbtype, rio *rdb, sds key, int dbid, int *error) {
                 return NULL;
             }
 
+            /* Also load the entry expiry */
+            long long itemexpiry = EXPIRY_NONE;
+            if (rdbtype == RDB_TYPE_HASH_2) {
+                itemexpiry = rdbLoadMillisecondTime(rdb, RDB_VERSION);
```

Comment:
A maliciously loaded Hash could have negative expirations with restore (It seems like we don't guard for that in the normal RDB load case either). I only briefly looked at the code, and I don't think it'll cause any issues, but it's probably worth erroring if the expiry was negative.

Reply:
perfect! I will do it